scripts/auto_dock.py

```diff
@@ -14,127 +14,191 @@
-        #Publishers
-        self.pub_aruco_detections_enable = rospy.Publisher('/aruco_detect/enable', Bool, queue_size=1, latch=True)
-        self.pub_cmd_vel = rospy.Publisher('/cmd_vel/auto_dock', TwistStamped, queue_size=1, latch=True)
-        self.pub_docking_state = rospy.Publisher('/auto_dock/state', String, queue_size=1, latch=True)
-        self.pub_action_state = rospy.Publisher('/auto_dock/action_state', String, queue_size=1, latch=True)
-        self.pub_docking_state.publish(self.docking_state_msg)
 
-        #Intialize Subscribers
-        self.sub_aruco_detect = rospy.Subscriber("/fiducial_transforms",FiducialTransformArray, self.aruco_detect_cb, queue_size=1)
-        self.sub_openrover_charging = rospy.Subscriber("rr_openrover_basic/charging",Bool, self.openrover_charging_cb, queue_size=1)
+        # Publishers
+        self.pub_aruco_detections_enable = rospy.Publisher(
+            "/aruco_detect/enable", Bool, queue_size=1, latch=True
+        )
+        self.pub_cmd_vel = rospy.Publisher(
+            "/cmd_vel/auto_dock", TwistStamped, queue_size=1, latch=True
+        )
+        self.pub_docking_state = rospy.Publisher(
+            "/auto_dock/state", String, queue_size=1, latch=True
+        )
+        self.pub_action_state = rospy.Publisher(
+            "/auto_dock/action_state", String, queue_size=1, latch=True
+        )
+        self.pub_docking_state.publish(self.docking_state_msg)
 
-        self.sub_undock = rospy.Subscriber("/auto_dock/undock", Bool, self.undock_cb, queue_size=1)
-        self.sub_cancel_auto_dock = rospy.Subscriber("/auto_dock/cancel", Bool, self.cancel_cb, queue_size=1)
-        self.sub_start = rospy.Subscriber("/auto_dock/dock", Bool, self.start_cb, queue_size=1)
-        #Setup timers
-        self.state_manager_timer = rospy.Timer(rospy.Duration(self.MANAGER_PERIOD), self.state_manage_cb, oneshot=False)
-        #self.docking_timer = rospy.Timer(rospy.Duration(self.MAX_RUN_TIMEOUT), self.docking_failed_cb, oneshot=True)
+        # Subscribers
+        self.sub_aruco_detect = rospy.Subscriber(
+            "/fiducial_transforms",
+            FiducialTransformArray,
+            self.aruco_detect_cb,
+            queue_size=1,
+        )
+        self.sub_openrover_charging = rospy.Subscriber(
+            "rr_openrover_basic/charging",
+            Bool,
+            self.openrover_charging_cb,
+            queue_size=1,
+        )
+
+        self.sub_undock = rospy.Subscriber(
+            "/auto_dock/undock", Bool, self.undock_cb, queue_size=1
+        )
+        self.sub_cancel_auto_dock = rospy.Subscriber(
+            "/auto_dock/cancel", Bool, self.cancel_cb, queue_size=1
+        )
+        self.sub_start = rospy.Subscriber(
+            "/auto_dock/dock", Bool, self.start_cb, queue_size=1
+        )
+
+        # service for determining whether we're in range of the charging station
+        rospy.wait_for_service("wibotic_connector_can/read_parameter")
```

Comment:
add a timeout, wrap in a try-except, report error

Reply:
I find using loginfo to show that the node is waiting for a service to be helpful during development and deployment.

Reply:
let me know what you guys think of the handling of this in the last commit

Reply:
Looks good in the newest commit!